releasenotes/notes/dag-oxide-60b3d7219cb21703.yaml

```diff
@@ -0,0 +1,49 @@
+---
+features_transpiler:
+  - |
+    The implementation of the :class:`.DAGCircuit` has been rewritten in Rust. This rewrite of
+    the Python class should be fully API compatible with the previous Python implementation of
+    the class. While the class was previously implemented using
+    `rustworkx <https://www.rustworkx.org/>`__ and it's underlying data graph structure existed
+    in Rust, the implementation of the class and all the data was stored in Python. This new
+    version of :class:`.DAGCircuit` stores a Rust native representation for all it's data and
+    is fully implemented in Rust. This new implementation should be more efficient in memory
+    usage as it compresses     the qubit and clbit representation for instructions at rest.
+    It also enables speed up for     transpiler passes as they can fully manipulate a
+    :class:`.DAGCircuit` from Rust.
+upgrade_transpiler:
+  - |
+    :class:`.DAGNode` objects (and its subclasses :class:`.DAGInNode`, :class:`.DAGOutNode`, and
+    :class:`.DAGOpNode`) are no longer returning references to the same underlying object from
+    :class:`.DAGCircuit` methods. This was never a guarantee before that all returned nodes would
```

Comment:
Something like this might be a bit clearer:
```yaml
  - |
    Repeated access to the data attached to a particular node on the DAG structure no longer
    returns references to the same :class:`.DAGNode` object (and subclasses :class:`.DAGInNode`,
    :class:`.DAGOutNode`, and :class:`.DAGOpNode`), but rather returns a new :class:`.DAGNode`
    instance upon each access. In the past the reference was often to the same object, although it
    was never guaranteed. Although returned objects are now distinct, comparing them with ``==`` or
    similar checks that rely on ``__eq__`` will return `True` if their content is the same.
```